mongodbatlas/config.go

```diff
@@ -50,29 +50,32 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 		return nil, diag.FromErr(err)
 	}
 
-	// Realm
-	optsRealm := []realm.ClientOpt{realm.SetUserAgent("terraform-provider-mongodbatlas/" + version.ProviderVersion)}
-	if c.BaseURL != "" {
-		optsRealm = append(optsRealm, realm.SetBaseURL(c.BaseURL))
-	}
-	authConfig := realmAuth.NewConfig(client)
-	token, err := authConfig.NewTokenFromCredentials(ctx, c.PublicKey, c.PrivateKey)
-	if err != nil {
-		return nil, diag.FromErr(err)
+	clients := &MongoDBClient{
+		Atlas: atlasClient,
 	}
 
-	clientRealm := realmAuth.NewClient(realmAuth.BasicTokenSource(token))
-	clientRealm.Transport = logging.NewTransport("MongoDB Realm", clientRealm.Transport)
+	// Realm
+	if len(c.PublicKey) > 0 && len(c.PrivateKey) > 0 {
+		optsRealm := []realm.ClientOpt{realm.SetUserAgent("terraform-provider-mongodbatlas/" + version.ProviderVersion)}
+		if c.BaseURL != "" {
+			optsRealm = append(optsRealm, realm.SetBaseURL(c.BaseURL))
+		}
+		authConfig := realmAuth.NewConfig(nil)
+		token, err := authConfig.NewTokenFromCredentials(ctx, c.PublicKey, c.PrivateKey)
+		if err != nil {
+			return nil, diag.FromErr(err)
+		}
 
-	// Initialize the MongoDB Realm API Client.
-	realmClient, err := realm.New(clientRealm, optsRealm...)
-	if err != nil {
-		return nil, diag.FromErr(err)
-	}
+		clientRealm := realmAuth.NewClient(realmAuth.BasicTokenSource(token))
+		clientRealm.Transport = logging.NewTransport("MongoDB Realm", clientRealm.Transport)
 
-	clients := &MongoDBClient{
-		Atlas: atlasClient,
-		Realm: realmClient,
+		// Initialize the MongoDB Realm API Client.
+		realmClient, err := realm.New(clientRealm, optsRealm...)
+		if err != nil {
+			return nil, diag.FromErr(err)
+		}
+
+		clients.Realm = realmClient
```

Comment:
This panics, needs to discuss a little bit further

Reply:
Pushed changes and now it won't panic